test/fixtures/detect-existing-browser-api-object/content.js

```diff
@@ -33,3 +33,22 @@ test("browser api object in background page", async (t) => {
     t.ok(!reply.windowBrowserIsUnchanged, "window.browser API object should have been defined by the polyfill");
   }
 });
+
+test("error types", async (t) => {
+  if (navigator.userAgent.includes("Firefox/")) {
+    try {
+      await browser.storage.sync.set({a: 'a'});
+      t.fail('It should throw when attempting to set an object over quota');
```

Comment:
Nit, the assertion message doesn't match the actual error scenario now, we could change it to `t.fail('storage.sync should throw when called from an extension with a temporary id')` or something like that.

(@fregante I can make the change and update this branch if you don't get to it first, in the end it is just a small nit related to an assertion message and so I don't mind if you prefer that I make the change and then land this on master).

Reply:
Minor tweak to one of the assertion messages applied in edaa539.